Fix evt.keyCode usage on IE6,7

fdaugan opened the pull request:
On IE7- keyCode property is does not correspond to the key code.
The fixed issue is relevant on IE7. The 't' key, keyCode on IE7 is equal to 116, so is mapped to F5 key.
So, using {999} patern allows digit and 't' key since isSpecialKey(key) return true.

fdaugan committed:
Fix evt.keyCode usage on IE6,7

On IE7- keyCode property is does not correspond to the key code.
The fixed issue is relevant on IE7. The 't' key, keyCode on IE7 is equal to 116, so is mapped to F5 key.
So, using {999} patern allows digit and 't' key since isSpecialKey(key) return true.

jaridmargolin commented:
I am in the midst (about 90% complete) of implementing automated cross browser testing using sauce labs.This will fix the issue with your failing tests, as well as confirm that this change does not effect other various browsers. Thank you for the PR.

fdaugan committed:
Fix evt.keyCode usage on IE6,7

I found the issue for keyPress/keyDown.
http://www.west-wind.com/WestwindWebToolkit/samples/Ajax/html5andCss3/keycodechecker.aspx

For keypress event does not map the keycode as keydown event.
Why are there 2 key event managements?

Any way, I've tested this PR on IE7, FF, Chrome and IE11.

fdaugan committed:
Update formatter.js

jaridmargolin commented:
Some fairly major changes have been pushed. Can you confirm this is still an issue?
